scout/adapter/mongo/query.py

```diff
@@ -474,63 +474,88 @@ def coordinate_filter(self, query, mongo_query):
+                                        {"end": {"$lte": end}},
+                                    ]
+                                },  # Case 4
+                            ]
+                        },
+                    ]
+                },
+                # Case chromosome != end_chrom, position matching 'chromosome'
+                {
+                    "$and": [
+                        {"chromosome": chrom},
+                        {"end_chrom": {"$ne": chrom}},
+                        {"position": {"$gte": start}},
+                        {"position": {"$lte": end}},
+                    ]
+                },
+                # Case chromosome != end_chrom, position matching 'end_chrom'
+                {
+                    "$and": [
+                        {"chromosome": {"$ne": chrom}},
+                        {"end_chrom": chrom},
+                        {"end": {"$gte": start}},
+                        {"end": {"$lte": end}},
+                    ]
+                },
+            ]
+        }
 
+    def sv_coordinate_query(self, query: dict) -> dict:
+        """Adds genomic coordinated-related filters to the query object
+        This method is called to build coordinate query for sv variants
         """
-        coordinate_query = None
-        chromosome_query = {"$or": [{"chromosome": query["chrom"]}, {"end_chrom": query["chrom"]}]}
-        if query.get("start") and query.get("end"):
-            # Query for overlapping intervals. Taking into account these cases:
-            # 1
-            # filter                 xxxxxxxxx
-            # Variant           xxxxxxxx
-
-            # 2
-            # filter                 xxxxxxxxx
-            # Variant                    xxxxxxxx
-
-            # 3
-            # filter                 xxxxxxxxx
-            # Variant                   xx
-
-            # 4
-            # filter                 xxxxxxxxx
-            # Variant             xxxxxxxxxxxxxx
-            position_query = {
-                "$or": [
-                    {"end": {"$gte": int(query["start"]), "$lte": int(query["end"])}},  # 1
-                    {
-                        "position": {
-                            "$lte": int(query["end"]),
-                            "$gte": int(query["start"]),
-                        }
-                    },  # 2
-                    {
-                        "$and": [
-                            {"position": {"$gte": int(query["start"])}},
-                            {"end": {"$lte": int(query["end"])}},
-                        ]
-                    },  # 3
-                    {
-                        "$and": [
-                            {"position": {"$lte": int(query["start"])}},
-                            {"end": {"$gte": int(query["end"])}},
-                        ]
-                    },  # 4
-                ]
+        if query.get("start") and query.get("end"):  # query contains full coordinates
```

Comment:
Hm, wouldn't that mean that e.g. start=0 is unused?

Reply:
It would probably happen that all variants on that chromosome are returned. Haven't changed this tho, it has been like this since forever. I'll fix it